v2/initiator.go

```diff
@@ -41,7 +43,9 @@ struct target_listing list_targets_wrapper(nfc_device *device, const nfc_modulat
 		targets = targets_tmp;
 		actual_count = nfc_initiator_list_passive_targets(device, nm, targets, targets_len);
 
-		// also covers the case where actual_count is an error
+		// negative actual_count is an error
+		if (actual_count < 0) break;
+
 		if (actual_count < targets_len) break;
 
 		targets_len += 16;
```

Comment:
This whole logic is fishy.  I think it should be something like this instead:

```suggestion
		// failure or array was long enough?
		if (actual_count < 0 || actual_count <= targets_len) break;

		// if not, try again with longer array
		targets_len = actual_count;
```